### PR TITLE
Fix the issue DEVTOOLEI-778

### PR DESCRIPTION
**Pull request by Bhashinee**
Issue : https://wso2.org/jira/browse/DEVTOOLEI-778

### CODE REVIEW

File: plugins/org.wso2.developerstudio.eclipse.gmf.esb.diagram/src/org/wso2/developerstudio/eclipse/gmf/esb/diagram/custom/deserializer/CacheMediatorDeserializer.java [hunk 1]
```diff
@@ -59,26 +60,27 @@ public CacheMediator createNode(IGraphicalEditPart part, AbstractMediator object
 			if (mediator.isCollector()) {
 				executeSetValueCommand(CACHE_MEDIATOR__CACHE_ACTION, CacheAction.COLLECTOR);
```
Comment (Bhashinee) [inline plugins/org.wso2.developerstudio.eclipse.gmf.esb.diagram/src/org/wso2/developerstudio/eclipse/gmf/esb/diagram/custom/deserializer/CacheMediatorDeserializer.java]: Formatting issues are fixed in : https://github.com/wso2/devstudio-tooling-esb/pull/272

File: plugins/org.wso2.developerstudio.eclipse.gmf.esb.persistence/src/org/wso2/developerstudio/eclipse/gmf/esb/internal/persistence/CacheMediatorTransformer.java [hunk 2]
```diff
@@ -77,19 +79,36 @@ public void transformWithinSequence(TransformationInfo information,
 		setCommonProperties(cacheMediator, visualCache);
 		{
 			if (visualCache.getCacheAction().getValue() == 0) {
-				cacheMediator.setHTTPMethodsToCache(visualCache.getCacheProtocolMethods().split(","));
 				cacheMediator.setProtocolType(visualCache.getCacheProtocolType().getLiteral());
-				cacheMediator.setResponseCodes(visualCache.getResponseCodes());
+				if (CachingConstants.HTTP_PROTOCOL_TYPE.equals(visualCache.getCacheProtocolType().getLiteral())) {
+					if (StringUtils.isNotBlank(visualCache.getCacheProtocolMethods())) {
```
Comment (prabushi) [inline plugins/org.wso2.developerstudio.eclipse.gmf.esb.persistence/src/org/wso2/developerstudio/eclipse/gmf/esb/internal/persistence/CacheMediatorTransformer.java]: Shall we format here? Spacing between line 83 and 84, and 84 and 85 differs.

File: plugins/org.wso2.developerstudio.eclipse.gmf.esb.persistence/src/org/wso2/developerstudio/eclipse/gmf/esb/internal/persistence/CacheMediatorTransformer.java [hunk 3]
```diff
@@ -77,19 +79,36 @@ public void transformWithinSequence(TransformationInfo information,
...
 				if (visualCache.getHashGenerator().equals("HTTP_REQUEST_HASH_GENERATOR")) {
-					org.wso2.carbon.mediator.cache.digest.DigestGenerator httpRequestHashGenerator = new org.wso2.carbon.mediator.cache.digest.HttpRequestHashGenerator();
-					cacheMediator.setDigestGenerator(httpRequestHashGenerator);
+				    httpRequestHashGenerator = new org.wso2.carbon.mediator.cache.digest.HttpRequestHashGenerator();
+				} else if (visualCache.getHashGenerator().toLowerCase().contains("requesthashgenerator")) {
+				    httpRequestHashGenerator = new org.wso2.carbon.mediator.cache.digest.REQUESTHASHGenerator();
+				} else if (visualCache.getHashGenerator().toLowerCase().contains("domhashgenerator")) {
+				    httpRequestHashGenerator = new org.wso2.carbon.mediator.cache.digest.DOMHASHGenerator();
+				} else {
+					throw new TransformerException("Digest generator not found");
```
Comment (prabushi) [inline plugins/org.wso2.developerstudio.eclipse.gmf.esb.persistence/src/org/wso2/developerstudio/eclipse/gmf/esb/internal/persistence/CacheMediatorTransformer.java]: Shall we format here too?

File: plugins/org.wso2.developerstudio.eclipse.gmf.esb.diagram/src/org/wso2/developerstudio/eclipse/gmf/esb/diagram/custom/deserializer/CacheMediatorDeserializer.java [hunk 4]
```diff
@@ -59,26 +60,27 @@ public CacheMediator createNode(IGraphicalEditPart part, AbstractMediator object
...
 			} else {
-				String[] methods = mediator.getHTTPMethodsToCache();
-				if (methods.length != 0 && !methods[0].isEmpty()) {
-					StringBuilder method = new StringBuilder();
-					for (int i = 0; i < methods.length - 1; i++) {
-						method.append(methods[i]).append(", ");
+				if (CachingConstants.HTTP_PROTOCOL_TYPE.equals(mediator.getProtocolType())) {
```
Comment (prabushi) [inline plugins/org.wso2.developerstudio.eclipse.gmf.esb.diagram/src/org/wso2/developerstudio/eclipse/gmf/esb/diagram/custom/deserializer/CacheMediatorDeserializer.java]: Shall we check for spacing after and inside else?

File: plugins/org.wso2.developerstudio.eclipse.gmf.esb.persistence/src/org/wso2/developerstudio/eclipse/gmf/esb/internal/persistence/CacheMediatorTransformer.java (hunk 2 above)
Comment (Bhashinee) [reply plugins/org.wso2.developerstudio.eclipse.gmf.esb.persistence/src/org/wso2/developerstudio/eclipse/gmf/esb/internal/persistence/CacheMediatorTransformer.java]: Formatting issues are fixed in : #272

File: plugins/org.wso2.developerstudio.eclipse.gmf.esb.persistence/src/org/wso2/developerstudio/eclipse/gmf/esb/internal/persistence/CacheMediatorTransformer.java (hunk 3 above)
Comment (Bhashinee) [reply plugins/org.wso2.developerstudio.eclipse.gmf.esb.persistence/src/org/wso2/developerstudio/eclipse/gmf/esb/internal/persistence/CacheMediatorTransformer.java]: Formatting issues are fixed in : #272

File: plugins/org.wso2.developerstudio.eclipse.gmf.esb.diagram/src/org/wso2/developerstudio/eclipse/gmf/esb/diagram/custom/deserializer/CacheMediatorDeserializer.java (hunk 4 above)
Comment (Bhashinee) [reply plugins/org.wso2.developerstudio.eclipse.gmf.esb.diagram/src/org/wso2/developerstudio/eclipse/gmf/esb/diagram/custom/deserializer/CacheMediatorDeserializer.java]: Formatting issues are fixed in : #272

**Pull request merged**